cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java

```diff
@@ -67,9 +82,153 @@ public class Constants {
     public static final String CRISTIN_BOT_FILTER_BYPASS_HEADER_VALUE =
         ENVIRONMENT.readEnv("CRISTIN_BOT_FILTER_BYPASS_HEADER_VALUE");
 
-    public enum QueryType {
-        QUERY_USING_GRANT_ID,
-        QUERY_USING_TITLE
-    }
+    private static final String PATTERN_IS_NUMBER6 = "[1-9]\\d*";
+    public static final String PATTERN_IS_DATE = "(\\d){4}-(\\d){2}-(\\d){2}[T]*[(\\d){2}:(\\d){2}:(\\d){2,6}Z]*";
+    private static final String PATTERN_IS_NON_EMPTY = ".+";
+    private static final String PATTERN_IS_TITLE = "^[æøåÆØÅ\\w-,\\. ]+$";
+    private static final String PATTERN_IS_LANGUAGE = "(en|nb|nn|\\,)+";
+    private static final String PATTERN_IS_STATUS = "(?i)CONCLUDED|ACTIVE|NOT[ +]*STARTED";
+    public static final String PATTERN_IS_URL =
+        "(http(s):\\/\\/.)[-a-zA-Z0-9@:%._\\+~#=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)";
+
+    public enum QueryParameterKey {
+        INVALID(null),
+        IDENTITY("identifier", null, PATTERN_IS_NON_EMPTY),
+        PATH_ORGANISATION("parent_unit_id", "organization", ORGANIZATION_IDENTIFIER_PATTERN,
+            ERROR_MESSAGE_INVALID_PATH_PARAMETER_FOR_ID_FOUR_NUMBERS, true),
+        PATH_PROJECT("projects", "project", PATTERN_IS_NUMBER6),
+        BIOBANK("biobank"),
+        FUNDING("funding"),
+        FUNDING_SOURCE("funding_source"),
+        GRANT_ID("project_code", null, PATTERN_IS_NUMBER6),
+        INSTITUTION("institution"),
+        LANGUAGE("lang", "language", PATTERN_IS_LANGUAGE),
+        LEVELS("levels", "depth", PATTERN_IS_NUMBER6),
+        NAME("name"),
+        PROJECT_APPROVAL_REFERENCE_ID("approval_reference_id"),
+        PROJECT_APPROVED_BY("approved_by"),
+        PROJECT_KEYWORD("keyword"),
+        PROJECT_ORGANIZATION("parent_unit_id","organization", PATTERN_IS_URL,
+            INVALID_URI_MESSAGE,true),
+        PROJECT_MANAGER("project_manager"),
+        PROJECT_MODIFIED_SINCE("modified_since", null, PATTERN_IS_DATE),
+        PROJECT_PARTICIPANT("participant"),
+        PROJECT_UNIT("unit"),
+        QUERY("query", null, PATTERN_IS_NON_EMPTY,
+              invalidQueryParametersMessage("query", ALPHANUMERIC_CHARACTERS_DASH_COMMA_PERIOD_AND_WHITESPACE),
```

Comment:
We should add support for more than alphanumeric etc, I heard frontend wants colon as well